app/src/on-chain.ts

```diff
@@ -130,4 +130,26 @@ export class OnChain {
 
     return { deploy, executionResults };
   }
+
+  async getDeploys() {
+    if (this.isSidecarRunning) {
+      try {
+        const {
+          status,
+          data: { data },
+        } = await this.sidecar.getDeploys();
+
+        this.checkSidecarStatus(status, data, () => this.getDeploys());
+
+        return data;
+      } catch (e) {
+        console.log("Error requesting deploys from sidecar.", e);
+
+        this.isSidecarRunning = false;
+        return this.getDeploys();
+      }
+    }
+
+    return "Sidecar was not able to fetch deploys data.";
```

Comment:
Should this throw an error rather than return a line of text?  I'm not 100% familiar with the new setup post NestJS

Reply:
I actually don't want it to throw an error. If you look at the rest of the file, you can see that if the sidecar isn't running then it defaults to fetching data via the rpc client. However, there is no such method to fetch `deploys` via the rpc client, so instead I just want to tell the FE that deploys can't be fetched. However, now that I think about it we should just be returning an empty array to match the expected data type. I'll update this now.

Reply:
Ahhh okay -- that makes sense!  I think the array would be the better approach :)